interfaces/Portalicious/src/app/services/auth/strategies/msal-auth/msal-auth.strategy.ts

```diff
@@ -98,7 +114,7 @@ export class MsalAuthStrategy implements IAuthStrategy {
     const subscription = this.msalService
       .handleRedirectObservable()
       .subscribe((data: AuthenticationResult | null) => {
-        if (!data) {
+        if (!data && !isIframed()) {
```

Comment:
Why this change?

Reply:
Apparently some handling of this already happens 'under the hood' in the popup subscribe. So in the popup scenario the data here is null. But it does have the token etc. so it's fine to continue.

Reply:
That makes me wonder whether we need the subscribe at all in that scenario, and whether we can just exit early from this function if `isIframed()`

Reply:
Something like

```ts
public handleAuthCallback(nextPageUrl: string) {
    if (isIframed()) {
      void this.refreshUserAndNavigate(nextPageUrl);
      return;
    }

    const subscription = this.msalService
      .handleRedirectObservable()
      .subscribe((data: AuthenticationResult | null) => {
        if (!data) {
          return;
        }
        subscription.unsubscribe();

        void this.refreshUserAndNavigate(nextPageUrl);
      });
  }
```